src/memory/primitives/felt.ts

```diff
@@ -0,0 +1,41 @@
+export class FeltError extends Error {}
+export class ConversionError extends Error {}
+
+export class Felt {
+  // TODO: should check for PRIME overflow.
+  // TODO: put private to make sure nothing is broken once this is added
+  private inner: bigint;
+  constructor(_inner: bigint) {
+    this.inner = _inner;
+  }
+
+  add(other: Felt): Felt {
+    return new Felt(this.inner + other.inner);
+  }
+
+  sub(other: Felt): Felt {
+    return new Felt(this.inner - other.inner);
+  }
+
+  getInner(): BigInt {
+    return this.inner;
```

Comment:
this is contradictory with the fact that it's `private`, why would we want this?